benchmark/src/main/java/com.google.cloud.bigquery/ConnImplBenchmark.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@Fork(value = 1)
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations = 0)
+@Measurement(iterations = 3)
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class ConnImplBenchmark {
+  @Param({"500000", "1000000", "10000000", "100000000"}) // 500K, 1M, 10M, and 100M
+  public int rowLimit;
+
+  private ConnectionSettings connectionSettingsReadAPIEnabled, connectionSettingsReadAPIDisabled;
+  private long numBuffRows = 100000L;
+  private final String DATASET = "bigquery_test_dataset";
+  private final String QUERY = "SELECT * FROM tlc_yellow_trips_2017_stephwang LIMIT %s";
+
+  @Setup
+  public void setUp() throws IOException {
+    java.util.logging.Logger.getGlobal().setLevel(Level.ALL);
+    ReadClientConnectionConfiguration clientConnectionConfiguration;
+
+    clientConnectionConfiguration =
+        ReadClientConnectionConfiguration.newBuilder()
+            .setTotalToPageRowCountRatio(10L)
+            .setMinResultSize(200000L)
+            .setBufferSize(numBuffRows)
+            .build();
+
+    connectionSettingsReadAPIEnabled =
+        ConnectionSettings.newBuilder()
+            .setDefaultDataset(DatasetId.of(DATASET))
+            .setNumBufferedRows(numBuffRows) // page size
+            .setPriority(
+                QueryJobConfiguration.Priority
+                    .INTERACTIVE) // DEFAULT VALUE - so that isFastQuerySupported returns false
```

Comment:
Yeah this is our way to avoid using`jobs.query` in this part of the benchmark since it does not support priority... I think as a future enhancement we can add the multiple modes but since it's an implementation detail that does not affect end user experience (we don't expect them to code against the implementation, just the interface), we can address that in the future.